src/psyclone/dynamo0p3.py

```diff
@@ -2863,6 +3963,88 @@ def deallocate(self, parent):
             parent.add(DeallocateGen(parent, sorted(func_space_var_names)))
 
 
+class DynBoundaryConditions(DynCollection):
+    '''
+    Manages declarations and initialisation of quantities required by
+    kernels that need boundary condition information.
+
+    :param node: the Invoke or Kernel stub for which we are to handle \
+                 any boundary conditions.
+    :type node: :py:class:`psyclone.dynamo0p3.DynInvoke` or \
+                :py:class:`psyclone.dynamo0p3.DynKern`
+    '''
+    # Define a BoundaryDofs namedtuple to help us manage the arrays that
+    # are required.
+    from collections import namedtuple
+    BoundaryDofs = namedtuple("BoundaryDofs", ["argument", "function_space"])
+
+    def __init__(self, node):
+        super(DynBoundaryConditions, self).__init__(node)
+
+        self._boundary_dofs = []
+        # Check through all the kernel calls to see whether any of them
+        # require boundary conditions. Currently this is done by recognising
+        # the kernel name.
+        for call in self._calls:
+            if call.name.lower() == "enforce_bc_code":
+                bc_fs = None
+                for fspace in call.arguments.unique_fss:
+                    if fspace.orig_name == "any_space_1":
+                        bc_fs = fspace
+                        break
+                if not bc_fs:
+                    raise GenerationError(
+                        "The enforce_bc_code kernel must have an argument on "
+                        "ANY_SPACE_1 but failed to find such an argument.")
+                farg = call.arguments.get_arg_on_space(bc_fs)
+                self._boundary_dofs.append(self.BoundaryDofs(farg, bc_fs))
+            elif call.name.lower() == "enforce_operator_bc_code":
+                # TODO check that the kernel only has one argument
```

Comment:
todo needs an issue.

Reply:
Rather than make yet another issue, I've added a check and associated test.